lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -172,12 +180,39 @@ void PDBImporterImpl::populateTypes() {
     return;
   }
 
+  // Those will be processed after all the types are visited.
+  DenseMap<TypeIndex, TypeIndex> ForwardReferencedTypes;
   PDBImporterTypeVisitor TypeVisitor(Importer.getModel(),
                                      InputFile->types(),
-                                     ProcessedTypes);
+                                     ProcessedTypes,
+                                     ForwardReferencedTypes,
+                                     *StreamTpiOrErr);
   if (auto Err = visitTypeStream(InputFile->types(), TypeVisitor)) {
     revng_log(DILogger, "Error during visiting types: " << Err);
     consumeError(std::move(Err));
+  } else {
+    // Post process the forward references.
+    for (auto &TypeInfo : ForwardReferencedTypes) {
+      auto ForwardDeclaration = TypeInfo.first;
+      auto OriginalType = TypeInfo.second;
+
+      if (ProcessedTypes.count(OriginalType) == 0
+          or ProcessedTypes.count(ForwardDeclaration) == 0)
+        continue;
+
+      auto ForwardModelType = ProcessedTypes[ForwardDeclaration];
+      auto ForwardModelTypeKey = ForwardModelType.get()->key();
+      auto OrignalModelType = ProcessedTypes[OriginalType];
+
+      auto CheckKey =
+        [&ForwardModelTypeKey](const model::TypePath &Reference) -> bool {
+        if (Reference.empty())
+          return false;
+
+        return ForwardModelTypeKey == Reference.getConst()->key();
+      };
+      Importer.getModel().replaceReferencesIf(OrignalModelType, CheckKey);
```

Comment:
Well, it does make sense! I will try to do that way.

Reply:
>You create a type for A (not sure which one).

I use an ugly hack for it: As a place holder type, I create a pointer to `void`, but the size of the pointer is the `sizeof` original type - and it can be an arbitrary `sizeof(a struct)`. It is important that I have the correct size for it, since I can still manually calculate the size of structs for the Model if it depends on the forward reference, and that is why I invoke `fixModel` and friends to clean up it later (since it is obviously wrong to have a pointer of size `X` in the Model).

Reply:
> size of the pointer is the sizeof original type

Mmh, this is not OK. Please see if what I suggested above can save us from doing this.
If not, let's figure out a proper solution.

In any case, when you leave such hacks, please leave a thorough comment explaining what's going on and why.

Reply:
@aleclearmind, I just pushed a new patch from @dwarfyresearch. Please take a look when you have a moment.

(`PDBImport: Handle forward referenced types` commit)

Reply:
Done.